dataprofiler/profilers/float_column_profile.py

```diff
@@ -96,25 +164,49 @@ def data_type_ratio(self):
         return None
 
     @classmethod
-    def _get_float_precision(cls, df_series_clean):
+    def _get_float_precision(cls, df_series_clean, sample_ratio=None):
         """
         Determines the precision of the numeric value.
         
         :param df_series_clean: df series with nulls removed, assumes all values
             are floats as well
         :type df_series_clean: pandas.core.series.Series
+        :param sample_ratio: Ratio of samples used for float precision
+        :type sample_ratio: float (between 0 and 1)
         :return: string representing its precision print format
         :rtype: int
         """
-        if not len(df_series_clean): return 0
+        len_df = len(df_series_clean)
+        if not len_df: return None
 
         # Lead zeros: ^[+-.0\s]+ End zeros: \.?0+(\s|$)
         # Scientific Notation: (?<=[e])(.*) Any non-digits: \D
         r = re.compile(r'^[+-.0\s]+|\.?0+(\s|$)|(?<=[e])(.*)|\D')
 
-        return float(df_series_clean.replace(
-            to_replace=r, value='').map(len).min())
+        # DEFAULT: Sample the dataset. If small use full dataset,
+        # OR 20k samples or 5% of the dataset which ever is larger.
+        # If user sets sample ratio, utilize their request
+        sample_size = min(len_df, max(20000, int(len_df * 0.05)))
+        if sample_ratio is not None and sample_ratio > 0:
+            sample_size = int(len_df * sample_ratio)
 
+        # length of sampled cells after all punctuation removed
+        len_per_float = df_series_clean.sample(sample_size).replace(
+            to_replace=r, value='').map(len)
+
+        # Determine statistics precision
+        precision_sum = float(len_per_float.sum())
+        subset_precision = {
+            'min': float(len_per_float.min()),
+            'max': float(len_per_float.max()),
+            'mean': precision_sum / sample_size,
+            'var': float(len_per_float.var()),
+            'sum': precision_sum,
```

Comment:
why don't use `float(len_per_float.sum())` to be consistent with other above calculations?

Reply:
oh, I see it's used for mean. Nvm.